parser_library/src/context/hlasm_context.cpp

```diff
@@ -58,6 +58,37 @@ hlasm_context::instruction_storage hlasm_context::init_instruction_map(id_storag
 }
 
 namespace {
+template<typename DATA>
+std::string make_string_with_leading_zeros(DATA data, size_t max_number_of_leading_zeros)
+{
+    std::string value = std::to_string(data);
+    if (auto value_len = value.size(); value_len < max_number_of_leading_zeros)
+        value.insert(0, max_number_of_leading_zeros - value_len, '0');
+
+    return value;
+}
+
+class sysstmt_macro_param_data : public macro_param_data_single_dynamic
+{
+    const performance_metrics& metrics;
+    mutable C_t data_ = "";
+
+public:
+    const C_t& get_dynamic_value() const override
+    {
+        size_t sysstmt = metrics.lines + metrics.macro_statements + 1;
```

Comment:
is this correct even for some of the more obscure cases like macro definition inside a macro call?

Reply:
You were right, nested macros have an effect on `SYSSTMT`

Reply:
The calculation of SYSSTMT now takes into account also `COPY` statements